chronicle/src/main/java/net/openhft/chronicle/tcp/SinkTcp.java

```diff
@@ -65,7 +84,8 @@ public boolean open(boolean blocking) throws IOException {
                 socketChannel.socket().setSendBufferSize(this.builder.sendBufferSize());
             }
 
-            super.setSocketChannel(socketChannel);
+            this.socketChannel = socketChannel;
+            this.tcpEventHandler = new TcpEventHandler(socketChannel, builder.tcpPipeline(sinkTcpHandler), sessionDetailsProvider, builder.sendBufferSize(), builder.receiveBufferSize());
```

Comment:
Not sure I follow you here.  Do you mean that by default the sink is supposed to be configured as blocking?